### ISSUE TITLE
Add more specific contexts

### ISSUE THREAD

**Issue by joshangell** [supercool/Pimp-My-Matrix#38]
Related to #28 and #33 there are some obvious contexts that need adding - feel free to comment to add more:

- [ ] Matrix fields in SuperTable fields - support all current contexts (Entry types, Category groups, Global sets, Users)
- [ ] Asset sources

**Comment by aladrach** [comment]
Were any of these fixes supposed to be included in the latest 2.1.3 update? Because it still appears that the nested fields with SuperTable and Matrix are not loading.

**Comment by joshangell** [comment]
@aladrach I'll take a look today for you, that may have got missed!

**Comment by joshangell** [comment]
@aladrach aha, I’ve found the scenario, it only happens when there is a block-specific field layout that contains a SuperTable / Matrix combo and only happens on load. You can get round it for now by either collapsing and expanding the affected blocks, or removing the field layout for that block. I've made a specific ticket if you want to follow that: #40

**Comment by dubcanada** [comment]
@joshangell Any idea of where to start looking at adding Supertable support for Pimp My Matrix? I need it for a site.

**Comment by joshangell** [comment]
@dubcanada sorry I haven’t actually looked into it - I think there will be a fair bit of js fangling to do. Feel free to take a shot and send a PR!

**Comment by fuelintegrated** [comment]
Looks like I'm trying to use the same setup as @aladrach but I don't have the problem of the fields not loading, but rather the group field layouts are completely ignored. E.g where in @aladrach 's screenshot the "Blocks" button inside the SuperTable row, even though I've configured that to appear, I just get the default Matrix row-of-buttons UI and no Content / Settings tabs on the Matrix fields.

**Comment by joshangell** [comment]
@fuelintegrated yep you can get around it either using @aladrach’s css fix (whatever that is) or removing the field layout for that block.

**Comment by fuelintegrated** [comment]
@joshangell I removed the field layout for the block, but the fields are still hidden :(

![screen shot 2016-05-26 at 11 41 07](https://cloud.githubusercontent.com/assets/3510875/15572182/d9f3d192-2336-11e6-9732-6fb3f775ae15.png)

**Comment by joshangell** [comment]
@fuelintegrated hmm, did you remove the global field layout for that block too? Or just the one specific to that entry type?

**Comment by fuelintegrated** [comment]
@joshangell There's still a field layout on the Column Module block type (you can see the Content / Settings tabs), would that be affecting it too?

**Comment by joshangell** [comment]
@fuelintegrated aha quite possibly - if removing that fixes it then please could you document that on the #40 thread

**Comment by fuelintegrated** [comment]
@joshangell That seems to solve it yeah, I'll document on #40

**Comment by aladrach** [comment]
@joshangell, @fuelintegrated, this is the override CSS:

```
.matrixblock .fields {
    display: block !important;
}
.matrixblock.collapsed .fields {
    display: none !important;
}
```
